### PR TITLE
[Enhancement] Support refresh hms table file meta in background

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Support refresh hms table file meta in background

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 13 (53.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/CacheUpdateProcessor.java|0|5|00.00%|[91, 160, 161, 179, 181]|
|:large_blue_circle:|com/starrocks/connector/hive/ConnectorTableMetadataProcessor.java|3|4|75.00%|[104]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -1683,10 +1683,16 @@ public class Config extends ConfigBase {
     public static long hive_max_split_size = 64L * 1024L * 1024L;
 
     /**
-     * Enable background refresh all hive external tables all partitions metadata on internal catalog.
+     * Enable background refresh all external tables all partitions metadata on internal catalog.
      */
     @ConfField
-    public static boolean enable_background_refresh_hive_metadata = false;
+    public static boolean enable_background_refresh_metadata = true;
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1689]: how about `enable_background_refresh_connector_metadata`

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/ConnectorTableMetadataProcessor.java [hunk 2]
```diff
@@ -57,11 +62,13 @@ public void unRegisterCacheUpdateProcessor(String catalogName) {
...
+        refreshRemoteFileExecutor = Executors.newFixedThreadPool(Config.background_refresh_file_metadata_concurrency,
+                new ThreadFactoryBuilder().setNameFormat("refresh-hive-remote-files-%d").build());
     }
 
     @Override
     protected void runAfterCatalogReady() {
-        if (!Config.enable_hms_events_incremental_sync && Config.enable_background_refresh_hive_metadata) {
+        if (!Config.enable_hms_events_incremental_sync) {
             refreshResourceHiveTable();
         }
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/ConnectorTableMetadataProcessor.java:74]: I can't find where use the registerTableInfo()

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 3]
```diff
@@ -1273,7 +1273,9 @@ private void startNonLeaderDaemonThreads() {
             metastoreEventsProcessor.start();
         }
 
-        connectorTableMetadataProcessor.start();
+        if (Config.enable_background_refresh_metadata) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1276]: It's better add backgroup options in the catalog properties

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 4]
```diff
@@ -1683,10 +1683,16 @@ public class Config extends ConfigBase {
...
+
+    /**
+     * Number of threads to refresh remote file's metadata concurrency.
+     */
+    @ConfField
+    public static int background_refresh_file_metadata_concurrency = 1;
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1695]: 1 is a little bit small.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 21 (61.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/CacheUpdateProcessor.java|0|5|00.00%|[91, 160, 161, 179, 181]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorInternalMgr.java|1|2|50.00%|[149]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|1|2|50.00%|[81]|
|:large_blue_circle:|com/starrocks/connector/hive/ConnectorTableMetadataProcessor.java|3|4|75.00%|[104]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnectorInternalMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java]: Done

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/ConnectorTableMetadataProcessor.java [hunk 5]
```diff
@@ -57,11 +62,13 @@ public void unRegisterCacheUpdateProcessor(String catalogName) {
 
     public ConnectorTableMetadataProcessor() {
         super(ConnectorTableMetadataProcessor.class.getName(), Config.background_refresh_metadata_interval_millis);
+        refreshRemoteFileExecutor = Executors.newFixedThreadPool(Config.background_refresh_file_metadata_concurrency,
+                new ThreadFactoryBuilder().setNameFormat("refresh-hive-remote-files-%d").build());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/ConnectorTableMetadataProcessor.java:66]: background-refresh-xxxx ?

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 6]
```diff
@@ -1273,7 +1273,9 @@ private void startNonLeaderDaemonThreads() {
             metastoreEventsProcessor.start();
         }
 
-        connectorTableMetadataProcessor.start();
+        if (Config.enable_background_refresh_connector_metadata) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1276]: @DorianZheng Please double check. You are responsible for this configuration once online

**Review by stephen-shelby — APPROVED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 21 (61.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/CacheUpdateProcessor.java|0|5|00.00%|[91, 160, 161, 179, 181]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorInternalMgr.java|1|2|50.00%|[149]|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|1|2|50.00%|[81]|
|:large_blue_circle:|com/starrocks/connector/hive/ConnectorTableMetadataProcessor.java|3|4|75.00%|[104]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnectorInternalMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveConnector.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Comment by Youngwb** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**